swarm/api/http/server.go

```diff
@@ -935,3 +947,28 @@ func (lrw *loggingResponseWriter) WriteHeader(code int) {
 func isDecryptError(err error) bool {
 	return strings.Contains(err.Error(), api.ErrDecrypt.Error())
 }
+
+// periodicTagTrace queries the tag every 2 seconds and logs its state to the span
+func periodicTagTrace(tags *chunk.Tags, tagUid uint32, q chan struct{}, sp opentracing.Span) {
+	f := func() {
+		tag, err := tags.Get(tagUid)
+		if err != nil {
+			log.Warn("got an error getting tag", "tagUid", tagUid, "err", err)
+		}
+
+		sp.LogFields(olog.String("tag state", fmt.Sprintf("split=%d stored=%d seen=%d synced=%d", tag.Get(chunk.StateSplit), tag.Get(chunk.StateStored), tag.Get(chunk.StateSeen), tag.Get(chunk.StateSynced))))
+	}
+
+	for {
+		select {
+		case <-q:
+			f()
+
+			return
+		default:
+			f()
+
+			time.Sleep(2 * time.Second)
```

Comment:
because we want a delay between calls that log the state of the upload tag. We don't want to have a million log lines in the tracing system, but rather one line every two seconds.